sites/website/src/docs/resources/cheat-sheet.md

```diff
@@ -0,0 +1,711 @@
+| index | The index of the current item when inside a repeat block (opt-in). |
+| length | The length of the array when inside a repeat block (opt-in). |
+| isEven | True if the index of the current item is even when inside a repeat block (opt-in). |
+| isOdd | True if the index of the current item is odd when inside a repeat block (opt-in). |
+isFirst | True if the current item is first in the array inside a repeat block (opt-in). |
+isInMiddle | True if the current item is somewhere in the middle of the array inside a repeat block (opt-in). |
+isLast | True if the current item is last in the array inside a repeat block (opt-in). |
+
+---
+
+### [Styles](https://www.fast.design/docs/fast-element/leveraging-css#basic-styles)
+
+`FASTElement` provides a css tagged template helper that allows for the creation of `ElementStyles`.
+
+Adding CSS to a `FASTElement`:
+
+```ts
+import { FASTElement, customElement } from '@microsoft/fast-element';
+import { css } from "@microsoft/fast-element";
+import { disabledOpacity } from "../design-tokens";
+
+const styles = css`
+  :host([disabled])] {
+    opacity: ${disabledOpacity};
+  }
+`;
+
+@customElement({
+  styles
+})
+export class MyElement extends FASTElement {}
+```
+
+---
+#### [Composing styles](https://www.fast.design/docs/fast-element/leveraging-css#composing-styles)
+
+`ElementStyles` can be composed with other styles:
+
+```ts
+import { normalize } from './normalize';
+
+const styles = css`
+  ${normalize}
+  :host {
+    ...
+  }
+`;
+```
+
+---
+
+#### [Partial CSS](https://www.fast.design/docs/fast-element/leveraging-css#partial-css)
+
+Use the `cssPartial` tagged template literal to create reusable blocks of partial CSS:
+
+```ts
+import { css, cssPartial } from "@microsoft/fast-element";
+
+const partial = cssPartial`color: red;`;
+const styles = css`:host{ ${partial} }`;
+```
+---
+### [CSSDirective](https://www.fast.design/docs/fast-element/leveraging-css#cssdirective)
+
+To create a `CSSDirective`, import and extend `CSSDirective` from `@microsoft/fast-element`:
+
+```ts
+import { CSSDirective }  from "@microsoft/fast-element"
+
+class RandomWidth extends CSSDirective {}
+```
+
+#### [createCSS method](https://www.fast.design/docs/fast-element/leveraging-css#createcss):
+
+`CSSDirective` has a `createCSS()` method that returns a string to be interpolated into an `ElementStyles`:
+
+```ts
+class RandomWidth extends CSSDirective {
+  createCSS() {
+    return "width: var(--random-width);"
+  }
+}
+```
+
+#### [createBehavior method](https://www.fast.design/docs/fast-element/leveraging-css#createbehavior):
+
+The `createBehavior()` method can be used to create a `Behavior` that is bound to the element using the `CSSDirective`:
+
+```ts
+class RandomWidth extends CSSDirective {
+  private property = "--random-width";
+  createCSS() {
+    return `width: var(${this.property});`
```

Comment:
This example produces an invalid property

Reply:
@radium-v Thank you - updated!

Reply:
@radium-v Thank you - updated!  

Would you mind reviewing the updated example?

Also, the code example was taken from the docs.  https://www.fast.design/docs/fast-element/leveraging-css

@EisenbergEffect Would it make sense to update here too in the context of the entire page's content?

![Screen Shot 2021-12-15 at 11 28 38 AM](https://user-images.githubusercontent.com/35910788/146252367-739bcbe6-1359-4f4b-bbf3-5c8f03693853.png)

Reply:
Yes, please.

Reply:
@EisenbergEffect Thank you - updated!